lib/kube/deploy.sh

```diff
@@ -6,8 +6,41 @@
 # optional - DOPPLER_TOKEN, DOPPLER_TOKEN_SECRET_NAME, DOPPLER_MANAGED_SECRET_NAME, KUBE_LABELS
 
 # custom vars
-# deployment id based on current timestamp
-export KUBE_DEPLOY_ID=$(( ($(date +%s) + 500) % 1000000 ))
+# Function to get the highest priority class value
+get_highest_priority_from_pods() {
+    # Get priority values from running pods in the specified namespace
+    priority_values=$(kubectl get pods -n "$KUBE_NS" -o custom-columns=PRIORITY:.spec.priority --no-headers 2>/dev/null)
+    
+    if [ -z "$priority_values" ]; then
+        echo 0
+        return
+    fi
+
+    # Extract the highest priority value
+    echo "$priority_values" | sort -n | tail -1
+}
+
+# Get the current highest priority value from pods
+HIGHEST_PRIORITY=$(get_highest_priority_from_pods)
+
+# Generate a base deployment ID based on the current timestamp, ensuring it's within the range 1 to 1000000000
+BASE_ID=$(( ( $(date +%s) % 1000000000 ) + 1 ))
+
+# Set the deployment ID initially to the base ID
+KUBE_DEPLOY_ID=$BASE_ID
+
+# If the KUBE_DEPLOY_ID is not greater than the highest priority class, adjust it
+if [ -n "$HIGHEST_PRIORITY" ]; then
+    if [ "$KUBE_DEPLOY_ID" -le "$HIGHEST_PRIORITY" ]; then
+        KUBE_DEPLOY_ID=$((HIGHEST_PRIORITY + 1))
+        if [ "$KUBE_DEPLOY_ID" -gt 1000000000 ]; then
+            KUBE_DEPLOY_ID=1000000000
```

Comment:
IMO by any chance if `HIGHEST_PRIORITY` value is `1000000000` then for the rest of the deployments the deployment id will be `1000000000`. Like:
- If the value of `HIGHEST_PRIORITY` is `1000000000` then `KUBE_DEPLOY_ID` will also be `1000000000`. The same is true for the next iterations too cause in the next iteration the `HIGHEST_PRIORITY` will also be `1000000000`(the same value has been already initialized to two deployments) and thus `KUBE_DEPLOY_ID` will again equals to the same value. This will keep on happening for all the next iterations.

Correct me if I'm wrong but I think this should not happen.

Reply:
Yeah you are right. We also don't want to have same DEPLOY_ID to two or more node so this approach won't work.
Working on a different approach for now.
Thanks for putting this up.